src/kem.rs

```diff
@@ -31,6 +32,24 @@ pub enum Kem {
     X25519HkdfSha256 = 32,
 }
 
+impl FromStr for Kem {
+    type Err = IdLookupError;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        match &*s.to_lowercase() {
+            #[cfg(feature = "kem-dh-p256-hkdf-sha256")]
+            "dhp256hkdfsha256" | "dh-p256-hkdf-sha256" | "dhkem(p-256, hkdf-sha256)" => {
+                Ok(Self::DhP256HkdfSha256)
+            }
+            #[cfg(feature = "kem-x25519-hkdf-sha256")]
+            "x25519hkdfsha256" | "x25519-hkdf-sha256" | "dhkem(x25519, hkdf-sha256)" => {
```

Comment:
nit: the shorter nicknames for this and the P-256-based KEM are inconsistent on whether they include "dh-"

Reply:
What's your suggested resolution to this?

Reply:
I'd recommend dropping "dh" from "dhp256hkdfsha256". (I think it would be unusual to abbreviate DHKEM to DH, and more natural to omit the DHKEM part altogether)

Reply:
I wonder if it's worth raising this with the backing crate as well. I just cargo-culted the naming from [`hpke::kem::DhP256HkdfSha256`](https://docs.rs/hpke/latest/hpke/kem/struct.DhP256HkdfSha256.html)

Reply:
addressed in eab5053 — i also added versions that don't label the kdf, so `p256-sha256` and `x25519-sha256`. let me know if that's somehow inaccurate or if there are other uses of sha256 as part of these kems other than hkdf